src/components/views/elements/AppTile.tsx

```diff
@@ -130,10 +134,24 @@ export default class AppTile extends React.Component<IProps, IState> {
         }
 
         this.state = this.getNewState(props);
+        this.watchUserReady();
 
         this.allowedWidgetsWatchRef = SettingsStore.watchSetting("allowedWidgets", null, this.onAllowedWidgetsChange);
     }
 
+    private watchUserReady = () => {
+        if (OwnProfileStore.instance.isProfileInfoFetched) {
+            return;
+        }
+        OwnProfileStore.instance.once(UPDATE_EVENT, this.onUserReady);
+    };
+
+    private onUserReady = (): void => {
+        if (OwnProfileStore.instance.isProfileInfoFetched) {
```

Comment:
this is a given, the event only fires after a setState which always sets the fetchedAt so this if statement will never fail

Reply:
the update event is also emitted on a reset, where state is set to `{}`

Reply:
That happens during logout/soft-logout when everything is unmounted, but if you're keeping this if statement then going back to the `on` listener is better otherwise you would handle a `reset` case wrong (not that the component would remain rendered for it to matter but better be consistent with your approach I guess)

Reply:
I thought I would hit the reset case in the in-room registration flow i'm working on for a customer ticket but yeah, it looks like the `RoomView` + `AppTile` are remounted anyway. Will remove this check and stick with `once`